jcl/src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1216,16 +1216,15 @@ public final Module getUnnamedModule()
  * @exception	ClassNotFoundException
  *					If the class could not be found.
  */
-public Class<?> loadClass (String className) throws ClassNotFoundException {
-/*[IF Sidecar19-SE]*/
+public Class<?> loadClass(String className) throws ClassNotFoundException {
+/*[IF JAVA_SPEC_VERSION > 8]*/
 	if ((bootstrapClassLoader == null) || (this == bootstrapClassLoader)) {
 		Class<?> cls = VMAccess.findClassOrNull(className, bootstrapClassLoader);
-		if (cls == null) {
-			throw new ClassNotFoundException(className);
+		if (cls != null) {
+			return cls;
```

Comment:
It seems this would cause an endless loop for classes not found, since loadClass(String, boolean) calls this method via `bootstrapClassLoader.loadClass(className)`.

Reply:
Although loadClass(String, boolean) is implemented by the BootClassLoader, so this version isn't used. Which explains why we don't see any endless loop.

Reply:
When `cls` is not found by `bootstrapClassLoader` (`jdk.internal.loader.ClassLoaders$BootClassLoader`), `jdk.internal.loader.BuiltinClassLoader.loadClass(String cn, boolean resolve)` [1] is invoked instead which overrides `ClassLoader.loadClass(className, resolveClass)` in question. Hence there is no endless loop.